collectors/freeipmi.plugin/freeipmi_plugin.c

```diff
@@ -1422,20 +1439,194 @@ static size_t send_ipmi_sel_metrics_to_netdata(struct netdata_ipmi_state *state)
         );
     }
 
+    netdata_mutex_unlock(&stdout_mutex);
+
     return state->sel.events;
 }
 
+// ----------------------------------------------------------------------------
+
+static const char *get_sensor_state_string(struct sensor *sn) {
+    switch (sn->sensor_state) {
+        case IPMI_MONITORING_STATE_NOMINAL:
+            return "nominal";
+        case IPMI_MONITORING_STATE_WARNING:
+            return "warning";
+        case IPMI_MONITORING_STATE_CRITICAL:
+            return "critical";
+        default:
+            return "unknown";
+    }
+}
+
+static const char *get_sensor_function_priority(struct sensor *sn) {
+    switch (sn->sensor_state) {
+        case IPMI_MONITORING_STATE_WARNING:
+            return "warning";
+        case IPMI_MONITORING_STATE_CRITICAL:
+            return "critical";
+        default:
+            return "normal";
+    }
+}
+
+static void freeimi_function_sensors(const char *transaction, char *function __maybe_unused, int timeout __maybe_unused, bool *cancelled __maybe_unused) {
+    time_t expires = now_realtime_sec() + update_every;
+
+    BUFFER *wb = buffer_create(PLUGINSD_LINE_MAX, NULL);
```

Comment:
When you worked with `diskspace` plugin, you defined a buffer [type](https://github.com/netdata/netdata/blob/4c867cb3a61c6c2bcd32f82267b0e01327eba752/collectors/diskspace.plugin/plugin_diskspace.c#L641), but here you are not defining it. Is there any reason for this change? I would suggest we proceed with this definition.

Reply:
internal vs external plugin

Reply:
I see we [set content-type here](https://github.com/ilyam8/netdata/blob/b410b0de808091d38b8a0cb11602661e360152c9/collectors/freeipmi.plugin/freeipmi_plugin.c#L1610) in external plugins